### ISSUE TITLE
False positive: E539 when number as type parameter

### ISSUE THREAD

**Issue by mfalt** [tonyhffong/Lint.jl#232]
The following function
```julia
function f{N}(v::NTuple{N,Float64})
    for i = 1:N
        println(v[i])
    end
end
```
Gives the error `E539 i: assigning an error to a variable`, I assume this is because Lint doesn't recognize that `N` is a number. 
It is easy to verify by running
```julia
lintstr("function f{N}(v::NTuple{N,Float64}) for i = 1:N println(v[i]) end end")
```
but is is clearly not a real problem:
```julia
julia> f((1.0,3.0,2.0))
1.0
3.0
2.0
``` 

Verified on Julia Version 0.6.0-rc1.0

**Comment by TotalVerb** [comment]
Thanks for the report. Can you try `Pkg.checkout("Lint")` and see if the problem persists?

**Comment by mfalt** [comment]
Seems to work fine on master, thanks!
